### ISSUE TITLE
Several inconsistencies found, trying to configure pylsp

### ISSUE THREAD

**Issue by michaelaye** [jupyter-lsp/jupyterlab-lsp#1048]
This project's main README describes installing an example server like so, using the Advanced Settings Editor:

```
{
  "language_servers": {
    "pyls": {
      "serverSettings": {
        "pyls.plugins.pydocstyle.enabled": true,
        "pyls.plugins.pyflakes.enabled": false,
        "pyls.plugins.flake8.enabled": true
      }
    },
    "r-languageserver": {
      "serverSettings": {
        "r.lsp.debug": false,
        "r.lsp.diagnostics": false
      }
    }
  }
}
```
But when trying to use the settings for Language Servers, since recently called "Language Servers (Experimental)", it seems one first needs to create a new language server with the right  server name and manually activate it with a switch:

![image](https://github.com/jupyter-lsp/jupyterlab-lsp/assets/69774/8d755d3b-397b-47c5-9012-20c2417d191a)

and then opening the JSON Settings Editor, some of the JSON keywords seem to not match above given instructions:

![image](https://github.com/jupyter-lsp/jupyterlab-lsp/assets/69774/14f855cc-4826-49df-ae2e-f2b3b525a55d)

i.e.:

* `language_servers` is now `languageServers`
* and `pyls(p).serverSettings` is now `pyls(p).configuration`

(indicating the recommended replacement of pyls with pylsp).

Note, that this is just an observation of inconsistencies, while I tried to find out how to get a python lsp server running, which I haven't been able to do so far. Something has changed as I was able to do to so before, even wrote a medium post about it...

Another observation: Anytime I added a property using the GUI, the terminal log said:

```
2024-02-06 13:18:26,728 CET - WARNING - pylsp_jsonrpc.endpoint - Ignoring notification for unknown method $/setTrace
```

**Comment by krassowski** [comment]
> Language Servers (Experimental)

You do not have this extension installed. This is coming from JupyteLab 4.0 and is non functional without this extension (or another one). This extension replaces `Language Servers (Experimental)` with `Language Servers`

**Comment by krassowski** [comment]
Personally I am not happy that upstream includes `Language Servers (Experimental)` but no LSP features (which are in this extension) - if it has no features, I would prefer it to not show that settings page at all, but was told that I can just replace it downstream here so here we are. Also there is so much I can do convincing other to embrace LSP features upstream. It really comes down to users voice (and funding).

**Comment by krassowski** [comment]
Feel welcome to submit PR replacing `pyls` with `pylsp` in readme :)

**Comment by michaelaye** [comment]
Hm, now i don't have ANY language server settings shown anymore.. because I filled something into the experimental settings?

**Comment by michaelaye** [comment]
still have some stuff disabled, don't understand difference between @jupyterlab/lsp-extension (disabled) and @jupyter-lsp/jupyterlab-lsp  (enabled). still no settings there, hmpf.

**Comment by krassowski** [comment]
Can you try choosing and reopening settings tab? There is a known issue with some settings categories not showing up when loading for the first time with Settings editor open.

If it suits not help, is there anything in the browser console?

**Comment by michaelaye** [comment]
create new clean env and only installed jupyterlab-lsp and jupyter-lsp-python.

Settings still do not appear, but pylsp is being loaded.
This is in the console:

![image](https://github.com/jupyter-lsp/jupyterlab-lsp/assets/69774/8c06a733-d058-42b3-8ea6-80edfc36d4e7)

**Comment by michaelaye** [comment]
possibly more important context before that:

![image](https://github.com/jupyter-lsp/jupyterlab-lsp/assets/69774/d2187260-83f1-4412-b026-9b1ba3e1e613)

**Comment by krassowski** [comment]
Yes, this one suggests that you may have an invalid setting in `pylsp.configurationSources`. It should be either `pyflakes` of `pyflakes`:

https://github.com/jupyter-lsp/jupyterlab-lsp/blob/5afe3b406e6f5d19952c0a70ba859e7f8b0f3d61/python_packages/jupyter_lsp/jupyter_lsp/specs/config/pylsp.schema.json#L5-L14

Can you check/clear your settings? These are stored on disk in a path indicated by `jupyter lab path` under `@jupyter-lsp/jupyterlab-lsp/plugin.jupyterlab-settings`.

**Comment by krassowski** [comment]
Wait, you should not need to edit files on disk. You should be able to access the panel in the JSON Settings Editor

**Comment by michaelaye** [comment]
Sorry, i don't know how to get to the pylsp settings with the JSON Settings Editor when they don't appear in the list:

![image](https://github.com/jupyter-lsp/jupyterlab-lsp/assets/69774/b8106d2b-606b-4bb6-9706-ca7e390b0f49)

**Comment by krassowski** [comment]
Thank you, this is a bug. It should be there even if validation fails.

**Comment by krassowski** [comment]
What about editing the file on disk?

**Comment by michaelaye** [comment]
yes, i removed by settings folder and now the entry appears in the settings UI, thanks!
